src/shape.rs

```diff
@@ -348,3 +376,356 @@ mod test
+    {
+        RoundedRectangle {
+            top_left: rectangle.top_left,
+            bottom_right: rectangle.bottom_right,
+            radius
+        }
+    }
+
+    /// Returns a reference to the top left vertex.
+    #[inline]
+    pub const fn top_left(&self) -> &Vector2<T>
+    {
+        &self.top_left
+    }
+
+    /// Returns a reference to the bottom right vertex.
+    #[inline]
+    pub const fn bottom_right(&self) -> &Vector2<T>
+    {
+        &self.bottom_right
+    }
+}
+
+impl<T: Copy> RoundedRectangle<T>
+{
+    /// Returns a vector representing the top right vertex.
+    #[inline]
+    pub fn top_right(&self) -> Vector2<T>
+    {
+        Vector2::new(self.bottom_right.x, self.top_left.y)
+    }
+
+    /// Returns a vector representing the bottom left vertex.
+    #[inline]
+    pub fn bottom_left(&self) -> Vector2<T>
+    {
+        Vector2::new(self.top_left.x, self.bottom_right.y)
+    }
+
+    /// Returns the radius of the rounded corners.
+    #[inline]
+    pub fn radius(&self) -> T
+    {
+        self.radius
+    }
+
+    /// Returns the x value of the left border
+    #[inline]
+    pub fn left(&self) -> T
+    {
+        self.top_left.x
+    }
+
+    /// Returns the x value of the right border
+    #[inline]
+    pub fn right(&self) -> T
+    {
+        self.bottom_right.x
+    }
+
+    /// Returns the y value of the top border
+    #[inline]
+    pub fn top(&self) -> T
+    {
+        self.top_left.y
+    }
+
+    /// Returns the y value of the bottom border
+    #[inline]
+    pub fn bottom(&self) -> T
+    {
+        self.bottom_right.y
+    }
+
+    /// Returns a `Rectangle` representing the rectangle that encloses this
+    /// rounded rectangle.
+    #[inline]
+    pub fn as_rectangle(&self) -> Rectangle<T>
+    {
+        Rectangle::new(self.top_left, self.bottom_right)
+    }
+}
+
+impl<T: std::ops::Sub<Output = T> + Copy> RoundedRectangle<T>
+{
+    /// Returns the width of the rounded rectangle.
+    #[inline]
+    pub fn width(&self) -> T
+    {
+        self.bottom_right.x - self.top_left.x
+    }
+
+    /// Returns the height of the rounded rectangle.
+    #[inline]
+    pub fn height(&self) -> T
+    {
+        self.bottom_right.y - self.top_left.y
+    }
+
+    /// Returns a `Vector2` containing the width and height of the rounded
+    /// rectangle.
+    #[inline]
+    pub fn size(&self) -> Vector2<T>
+    {
+        Vector2::new(self.width(), self.height())
+    }
+}
+
+impl<T> RoundedRectangle<T>
+where
+    T: num_traits::AsPrimitive<f32>
+        + std::cmp::PartialOrd
+        + std::ops::Add<Output = T>
+        + std::ops::Sub<Output = T>
+        + std::ops::Mul<Output = T>
+        + std::ops::Neg<Output = T>
+        + std::ops::Div<Output = f32>
+        + std::ops::Div<f32, Output = T>
+{
+    /// Returns true if the specified point is inside this rounded rectangle.
+    /// Note: this is always inclusive, in contrast to the `contains` method
```

Comment:
We should keep the behaviour consistent (this should also be exclusive)

Reply:
I wasn't sure about how to handle the curves: there isn't a real "bottom" if the corner is a curve. By "exclusive" do you mean that instead of always inclusive it should be always exclusive? In which case it would be different from the Rectangle's contains() anyways